pkg/controller/petset/pet_set.go

```diff
@@ -222,12 +222,7 @@ func (psc *PetSetController) getPodsForPetSet(ps *apps.PetSet) ([]*api.Pod, erro
 	if err != nil {
 		return []*api.Pod{}, err
 	}
-	// TODO: Do we need to copy?
-	result := make([]*api.Pod, 0, len(pods))
-	for i := range pods {
-		result = append(result, &(*pods[i]))
```

Comment:
Sorry, I am afraid I didn't catch your meaning.

* I think petset controller only use the returned `Pods.List()` for reading and won't affect the elements of original list. So I remove the following code block directly.


```
    result := make([]*api.Pod, 0, len(pods))
	for i := range pods {
		result = append(result, &(*pods[i]))
	}
```

* Do we need to copy the elements of the original list? Like,
```
    result := make([]*api.Pod, 0, len(pods))
	for i := range pods {
        p := *pods[i]
		result = append(result, &p)
	}
	return result, nil
```

* I just can't understand the real purpose of copying pointers from an array to another, to protect the original list?